File: DP/DP/Factory/EnemyFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DP.Factory
{
    public class EnemyFactory : IFactory
    {
        public Enemy CreateEnemy(string enemyType)
        {
            if (enemyType == "P")
            {
                return new Plane();
            }
            else if (enemyType == "S")
            {
                return new Ship();
            }

            return null;
        }
    }
}

File: DPJava/src/com/company/KTU/Factory/Factory/EnemyFactory.java
package com.company.KTU.Factory.Factory;

public class EnemyFactory implements IFactory {
    @Override
    public Enemy createEnemy(String ui )
    {
        if(ui.equals("P")) {
            return new Plane("lektuvas", 17, 18);
        }
        if(ui.equals("S")) {
            return new Ship("laivas", 21, 23);
        }

        return null;

    }
}
